src/api/Register.d.ts

```diff
@@ -1,14 +1,15 @@
-import { String } from './String';
-import { Counter } from './Counter';
-import { List } from './List';
+import type { Str } from './Str';
+import type { Counter } from './Counter';
+import type { List } from './List';
+import type { Value } from './Value';
 
 /**
  * Register contains a single value or none at all.
  */
 export interface Register {
-  value(): Value; // Returns the value of the register.
-  setString(): String; // Sets the register to an empty string.
-  setCounter(): Counter; // Sets the register to a zeroed counter.
-  setList(): List; // Sets the register to an empty list.
-  clear(): void; // Resets the register to an empty state.
+  value(): Value | null; // Returns the value of the register.
+  setString(): Str; // Sets the register to an empty string.
+  setCounter?(): Counter; // Sets the register to a zeroed counter.
```

Comment:
For my education, what does it mean for an interface method to be optional? I've read [this](https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties), but it only refers to properties. Does it mean that a subclass doesn't need to implement this method?

Reply:
Precisely. In this case, I didn't want to implement `setCounter` at this point, since we're focusing on the string type.